docs/proposals/job_controller.md

```diff
@@ -0,0 +1,192 @@
+# Job Controller
+
+## Abstract
+A proposal for implementing a new controller - Job controller, which will be responsible
+for managing pods that require to run:
+* once on demand or at a specified point in time
+* repeatedly at a specified point in time.
+
+Several exiting issues were already created regarding that particular subject:
+* Distributed CRON jobs [#2156](https://github.com/GoogleCloudPlatform/kubernetes/issues/2156)
+* Job Controller [#1624](https://github.com/GoogleCloudPlatform/kubernetes/issues/1624)
+
+
+## Use Cases
+1. Ability to have a controller that will create a series of run-once pod(s).
+1. Ability to implement a MapReduce model.
+1. Ability to have a controller that will have an option to specify triggers on pod’s success/failure.
+1. Ability to specify number of restarts upon job failure.
+1. Ability to specify maximum running time for a job.
+1. Ability to create a repetitive job, eg. database backup, sending emails.
+1. Ability to create one time job to be run at a specified time, eg. system analysis.
+1. Ability to create a chain of jobs dependent one on another.
+
+
+## Motivation
+The main goal is to provide a new pod controller type, called Job, with the following
+characteristics:
+* Be able to start of one or several pods tracked as a single entity representing
+a one execution of this job. Final job execution status will reflect the completion
+status of all pods started for this run.
+* Be able to trigger a new job execution either on demand (triggered through server
+API by an external agent or by another subsystem), or periodically based on a time-based
+schedule defined in job specification part (using ISO8601 syntax notation).
+* Be able to restart a limited number of time failing containers of pods created
+by one job execution (using "Restart on failure" logic extended by a maximal restart count).
+* Be able to limit the execution time of pods created by a job.
+* Be able to provide status information about job executions, along with a (limited)
+history of previous job runs (containing start/stop times with completion exit code).
+* Be able to trigger additional jobs/actions based on a success/failure of a job.
+
+
+## Job
+Jobs, similarly to replication controller, are managed by a dedicated manager which
+takes care of all jobs existing in the system, performing the following tasks:
+* Executes a loop on the full list of defined jobs, calling on each a defined
```

Comment:
It might be sufficient to say here that the controller will be based on the upstream controller framework.